src/services/geometry/richgeo/IrtGeoDRICH.cc

```diff
@@ -186,7 +184,25 @@ void richgeo::IrtGeoDRICH::DD4hep_to_IRT() {
   // define the `cell ID -> pixel position` converter
   SetReadoutIDToPositionLambda();
 }
-
+TVector3 richgeo::IrtGeoDRICH::GetSensorSurface(CellIDType id){
+  auto nSectors= m_det->constant<int>("DRICH_num_sectors");
+  TVector3 normX(1, 0,  0); // normal vectors
+  TVector3 normY(0, -1, 0);
+  TVector3 sensorNorm;
+  auto cellMask = uint64_t(std::stoull(m_det->constant<std::string>("DRICH_cell_mask")));
+  auto sensor_info = this->m_sensor_info;
+  auto sID = id & cellMask;
+  auto sensor_info_it = sensor_info.find(sID);
+
+  auto sensor_obj = sensor_info_it->second;
+  auto normZdir = sensor_obj.surface_normZdir;
+
+  sensorNorm.SetX(static_cast<double>(normZdir.x()));
+  sensorNorm.SetY(static_cast<double>(normZdir.y()));
+  sensorNorm.SetZ(static_cast<double>(normZdir.z()));
+  //std::cout<<sensorNorm.X()<<std::endl;
+  return sensorNorm;
+}
```

Comment:
Remove unused variables and re-definitions, and protect against the case where the sensor cannot be found. This also switches to using `m_sensor_info.surface_offset.Unit()`, as suggested in other comment.
```suggestion
TVector3 richgeo::IrtGeoDRICH::GetSensorSurface(CellIDType id){
  TVector3 sensorNorm;
  auto cellMask = uint64_t(std::stoull(m_det->constant<std::string>("DRICH_cell_mask")));
  auto sID = id & cellMask;
  auto sensor_info_it = m_sensor_info.find(sID);
  if(sensor_info_it != m_sensor_info.end()) {
    auto sensor_obj = sensor_info_it->second;
    auto normZdir = sensor_obj.surface_offset.Unit();
    sensorNorm.SetX(static_cast<double>(normZdir.x()));
    sensorNorm.SetY(static_cast<double>(normZdir.y()));
    sensorNorm.SetZ(static_cast<double>(normZdir.z()));
  }
  else
    m_log->error("Cannot find sensor {} in IrtGeoDRICH::GetSensorSurface", id);

  return sensorNorm;
}
```